### DISCUSSION TITLE
Energymeter einbinden

### DISCUSSION THREAD

**Discussion by wrzlbrmft1** [goecharger/go-eCharger-API-v2#146] [General]
Hallo!
Ich habe bei mir im Haus bereits ein EnergyMeter verbaut. Die Datenpunkte ensprechen der SUNSPEC (Kostal Wechselrichter).
Kann der go-e-Charger diese Werte abfragen?

Zudem habe ich noch eine  zweite Anwendung: In der ist nicht ein klassisches EM verbaut, sondern ein Shelly 3EM. Kann dieser direkt mit dem go-e-Charger kommunizieren?

Gibt es eine Spec, welche Energymeter mit dem go-e-charger funktionieren?
Wie offen ist die Spezfikation für GRID, damit ich mir notfalls eine Schnittstelle generieren kann, damit der go-e-charger weiss, ob ein PV-Überschuss da ist?

Sonnige Grüße!

**Comment by 0xFEEDC0DE64** [comment]
Der Charger kann selbst gar keine meter auslesen, der kommende go-eController kann seine Leistungswerte direkt in den Charger pushen, für Entwicklungszwecke kann man GRID PV und BATTERY Leistungswerte über unsere API hineinschicken.

api key "ids"
bsp. value:
```
{"pGrid": 1000., "pPv": 1000., "pAkku": 1000.}
```

achtung:
überschuss-laden und automatische phasenumschaltung sollte über die app aktiviert werden (wären sonst die api keys lmo=4 fup=true psm=0).

**Reply by wrzlbrmft1** [reply]
. . .  danke, genau das hab ich gesucht.
Wie wichtig sind an dieser Stelle die Werte "pPV" und "pAkku" ?
Ich hab bei der Anwendung leider "nur" den Wert für "pGrid", die Wechselrichter (-> plural) sind so unkommunikativ, dass ich von denen keine Werte bekomme.
Mercy vielmals!

**Reply by JoeSchwettmann** [reply]
> Der Charger kann selbst gar keine meter auslesen, der kommende go-eController kann seine Leistungswerte direkt in den Charger pushen, für Entwicklungszwecke kann man GRID PV und BATTERY Leistungswerte über unsere API hineinschicken.
> 
> api key "ids" bsp. value:
> 
> ```
> {"pGrid": 1000., "pPv": 1000., "pAkku": 1000.}
> ```
> 
> achtung: überschuss-laden und automatische phasenumschaltung sollte über die app aktiviert werden (wären sonst die api keys lmo=4 fup=true psm=0).

Ich habe es auch mal versucht, aber mit der aktuelle Firmware 56.8 und auch die Beta 57.0 hat es leider nicht sauber funktioniert. Er fängt zwar an zu laden, aber er bleibt immer bei der aktuellen Ampere Zahl (amp) stehen. Er regelt also nicht selbständig hoch. Egal wie ich pGrid verändere. Muss noch etwas anderes eingestellt sein? (Imo=4 fup=true und psm=0 sind gesetzt)

**Reply by JoeSchwettmann** [reply]
Hardware ist einmal V3 und einmal V4, beiden verhalten sich gleich.

**Reply by JoeSchwettmann** [reply]
Jetzt hat es doch endlich funktioniert. Bei mir stand amp auf 6, es muss wirklich auf 16 stehen, dann regelt er auch höher.
